From e60f53b2d6edd275d135ae14b9b394a720415489 Mon Sep 17 00:00:00 2001
From: CuriousNom <aryanbiswas7585@gmail.com>
Date: Wed, 19 Mar 2025 11:54:00 +0530
Subject: [PATCH] scripts: dtc: Avoid assertion failure in
 get_node_by_phandle()

Fixes an issue where `dtc` would crash due to an assertion failure
when `generate_fixups` is disabled. Instead of asserting, log a warning
and continue execution.

This prevents core dumps and ensures smoother compilation.

Signed-off-by: CuriousNom <aryanbiswas7585@gmail.com>
---
 scripts/dtc/livetree.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/scripts/dtc/livetree.c b/scripts/dtc/livetree.c
index 6e4c367f54b3..a974137f1b2d 100644
--- a/scripts/dtc/livetree.c
+++ b/scripts/dtc/livetree.c
@@ -562,7 +562,9 @@ struct node *get_node_by_phandle(struct node *tree, cell_t phandle)
 	struct node *child, *node;
 
 	if ((phandle == 0) || (phandle == -1)) {
-		assert(generate_fixups);
+		if (!generate_fixups) {
+                    fprintf(stderr, "Warning: generate_fixups is disabled. Continuing without fixups.\n");
+                }
 		return NULL;
 	}
 

